TrustKit/Pinning/pinning_utils.m

```diff
@@ -0,0 +1,70 @@
+/*
+
+ pinning_utils.m
+ TrustKit
+
+ Copyright 2023 The TrustKit Project Authors
+ Licensed under the MIT license, see associated LICENSE file for terms.
+ See AUTHORS file for the list of project authors.
+
+ */
+
+#import "pinning_utils.h"
+#include <dlfcn.h>
+#include "TargetConditionals.h"
+
+bool evaluateTrust(SecTrustRef serverTrust, SecTrustResultType *trustResult, NSError **error) {
```

Comment:
Let's remove this wrapper as the two locations that need to evaluate a trust object require slightly different error & result checking.

Reply:
I removed most of the logic, especially that calling `SecTrustGetTrustResult()`, but kept a thin wrapper for convenience around `SecTrustEvaluateWithError()` that re-interprets the `CFErrorRef` as an `NSError` (with an ownership transfer). It's now named `evaluateCertificateChainTrust()` to emphasize the nature of the trust being evaluated.